File: Programmers/Level2/230111/app/src/main/java/FindPrimeNumber.java
import java.util.HashSet;
import java.util.Iterator;

public class FindPrimeNumber {
    HashSet<Integer> hashSet = new HashSet<>();

    public int solution(String numbers) {
        int answer = 0;

        recursive("", numbers);

        Iterator<Integer> iterator = hashSet.iterator();

        while (iterator.hasNext()) {
            if(isPrime(iterator.next())) {
                answer += 1;
            }
        }

        return answer;
    }

    private void recursive(String combination, String numbers) {
        if(!combination.equals("")) {
            hashSet.add(Integer.parseInt(combination));
        }

        for(int i = 0; i < numbers.length(); i += 1) {
            recursive(combination + numbers.charAt(i),
                numbers.substring(0, i) + numbers.substring(i + 1));
        }
    }

    private boolean isPrime(int number) {
        if(number == 0 || number == 1) {
            return false;
        }

        int size = (int) Math.sqrt(number);

        for(int i = 2; i <= size; i += 1) {
            if(number % i == 0) {
                return false;
            }
        }

        return true;
    }
}

File: Programmers/Level1/220911/JavaScript/FindPrimeNumber.js
export default function solution(n) {
  const answer = 0;

  return answer;
}

// export function numberArray(n) {
//   const numbers = [];

//   for (let i = 2; i <= n; i += 1) {
//     numbers.push(i);
//   }

//   return numbers;
// }

// export function solution(n) {
//   const answer = {
//     result: 0,
//   };

//   const numbers = numberArray(n);

//   let count = 0;

//   for (const number of numbers) {
//     for (let i = 2; i < number; i += 1) {
//       if (number % i === 0) {
//         count += 1;
//         break;
//       }
//     }

//     if (count === 0) {
//       answer.result += 1;
//     }
//     count = 0;
//   }

//   return answer.result;
// }
